### PR TITLE
added IsStudent annotation

### PR DESCRIPTION
**Pull request by teodorstrut**
i have added the IsStudent annotation, and the preHandle method fot the annotation

### CODE REVIEW

File: backend/src/main/java/ubb/courses/backend/services/security/SecurityService.java [hunk 1]
```diff
@@ -39,4 +39,12 @@ public boolean isTeacher() {
                 authority -> authority.getAuthority().equals(AuthorityType.TEACHER.name())
         ).count() == 1;
     }
+
+    @Override
+    public boolean isStudent() {
+        val authorities = this.getAuthorities();
+        return authorities.stream().filter(
+                authority -> authority.getAuthority().equals(AuthorityType.STUDENT.name())
+        ).count() == 1;
```
Comment (teodorvoicencu) [inline backend/src/main/java/ubb/courses/backend/services/security/SecurityService.java]: I think .anyMatch would be a better approach than filtering and counting. Could you please change this everywhere ?

File: backend/src/main/java/ubb/courses/backend/security/AuthorityInterceptor.java [hunk 2]
```diff
@@ -39,6 +40,17 @@ public boolean preHandle(HttpServletRequest request, HttpServletResponse respons
...
 
+        IsStudent isStudent = handlerMethod.getMethodAnnotation(IsStudent.class);
+        
+        if (isStudent != null) {
+            if (securityService.isStudent()) {
+                return true;
+            } else {
+                response.setStatus(HttpStatus.FORBIDDEN.value());
+                return false;
+            }
```
Comment (teodorvoicencu) [inline backend/src/main/java/ubb/courses/backend/security/AuthorityInterceptor.java]: Else condition is unnecessary. Please change it everywhere.

File: backend/src/main/java/ubb/courses/backend/services/security/SecurityService.java (hunk 1 above)
Comment (teodorstrut) [reply backend/src/main/java/ubb/courses/backend/services/security/SecurityService.java]: Done ^^

File: backend/src/main/java/ubb/courses/backend/security/AuthorityInterceptor.java (hunk 2 above)
Comment (teodorstrut) [reply backend/src/main/java/ubb/courses/backend/security/AuthorityInterceptor.java]: Done ;)

**Review by teodorvoicencu — APPROVED** [review]
LGTM

**Pull request merged**